mitxgraders/comparers/linear_comparer.py

```diff
@@ -139,28 +147,54 @@ def __init__(self, config=None, **kwargs):
         'linear': get_linear_fit_error,
     }
 
+    @staticmethod
+    def check_comparing_zero(comparer_params_evals, student_evals, tolerance):
+        """
+        Check whether student input is nearly zero, or author input is exactly zero
+        """
+        student_zero = all([
+            is_nearly_zero(x, tolerance, reference=y)
+            for x, y in zip(comparer_params_evals, student_evals)
+        ])
+        expected_zero = all(x == 0.0 for [x] in comparer_params_evals)
+        return student_zero or expected_zero
+
+
+    def get_valid_modes(self, is_comparing_zero):
+        """
+        Returns a copy of self.modes, first removing 'proportional' and 'linear'
+        when is_comparing_zero is truthy.
+        """
+        remove_if_zero = set(['linear', 'proportional'])
+        invalid = lambda mode: is_comparing_zero and mode in remove_if_zero
+        return tuple(mode for mode in self.modes if not invalid(mode) )
```

Comment:
Minor linting at the end here. Can I suggest avoiding the double negative and doing an "allowed_if_zero" and "valid" instead?

Reply:
👍 new version is easier to read, I think